src/feature/client/entrynodes.c

```diff
@@ -1849,6 +1868,9 @@ make_guard_confirmed(guard_selection_t *gs, entry_guard_t *guard)
 
   guard->confirmed_idx = gs->next_confirmed_idx++;
   smartlist_add(gs->confirmed_entry_guards, guard);
+  /** The confirmation ordering might not be the sample ording. We need to
+   * reorder */
```

Comment:
Let's document this new requirement in the documentation for confirmed_entry_guards.

Reply:
ok, done in struct guard_selection_t

Reply:
Hmm, `compare_guards_by_sampled_idx()` only talks about sampled set but here we do it in the confirmed set. So perhaps we should update documentation?